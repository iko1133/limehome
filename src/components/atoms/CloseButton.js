import React from "react";
import styled from "styled-components/native";

const SafeArea = styled.SafeAreaView``;

const Container = styled.TouchableOpacity`
  align-items: center;
  background-color: ${({ theme }) => theme.BACKGROUND_COLOR};
  border-radius: 6px;
  height: 35px;
  justify-content: center;
  width: 35px;
`;

const CloseButtonX = styled.Text`
  font-size: 20px;
`;

/**
 * Default close button (with safeAreaView in mind)
 * @param {function} onPress - a callback called when button is pressed
 * @param {object} style - styling of the button container
 * @param {object} buttonStyle - styling of the button itself
 */

const CloseButton = ({ onPress, style, buttonStyle }) => {
  return (
    <SafeArea style={style}>
      <Container onPress={onPress} activeOpacity={0.8} style={buttonStyle}>
        <CloseButtonX>X</CloseButtonX>
      </Container>
    </SafeArea>
  );
};

export default CloseButton;
