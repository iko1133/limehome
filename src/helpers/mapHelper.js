// Copies given object, except, switches names of the location property's properties for the
// marker objects to function properly
// lgn to longitude
// lat to latitude
export const fixObjectLocationProperty = (object) => ({
  ...object,
  location: {
    ...object.location,
    longitude: object.location?.lng,
    latitude: object.location?.lat,
  },
});
