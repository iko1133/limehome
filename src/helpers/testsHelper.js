import { ThemeProvider } from "styled-components/native";

import { lightTheme } from "styles/theme";

/**
 * Wraps given children into a ThemeProvider with lightTheme set as the main theme
 * @param {object} children - children to be rendered inside ThemeProvider
 */
export const WrapInThemeProvider = (children) => (
  <ThemeProvider theme={lightTheme}>{children}</ThemeProvider>
);
