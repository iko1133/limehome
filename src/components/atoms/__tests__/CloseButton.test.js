import React from "react";
import renderer from "react-test-renderer";

import { WrapInThemeProvider } from "helpers/testsHelper";
import CloseButton from "../CloseButton";

describe("Testing CloseButton", () => {
  it("Renders close button", () =>
    expect(
      renderer.create(WrapInThemeProvider(<CloseButton />))
    ).toMatchSnapshot());
});
