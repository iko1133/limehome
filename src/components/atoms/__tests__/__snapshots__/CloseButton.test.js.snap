// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CloseButton Renders close button 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {},
    ]
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F7F1E9",
        "borderRadius": 6,
        "height": 35,
        "justifyContent": "center",
        "opacity": 1,
        "width": 35,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontSize": 20,
          },
        ]
      }
    >
      X
    </Text>
  </View>
</RCTSafeAreaView>
`;
